rules/utils_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/testutil"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func TestGetHTTPRulesEmpty(t *testing.T) {
+	tmpl := `
+	syntax = "proto3";
+
+	import "google/api/annotations.proto";
+
+	service Library {
+		rpc GetBook(GetBookRequest) returns (Book) {
+			{{ .Option }}
+		}
+	}
+
+	message GetBookRequest {
+		string name = 1;
+	}
+
+	message Book {}`
+
+	tests := []struct {
+		Option    string
+		ruleCount int
+	}{
+		{"", 0},
+		{"option (google.api.http) = { get: \"/v1/publishers/*/books/*\" };", 1},
+		{"option (google.api.http) = { get: \"/v1/publishers/*/books/*\" additional_bindings: { get: \"/v1/authors/*/books/*\" } };", 2},
+	}
+
+	for _, test := range tests {
+		fdp := testutil.MustCreateFileDescriptorProto(
+			t,
+			testutil.FileDescriptorSpec{
+				AdditionalProtoPaths: []string{testdatadir("api-common-protos")},
+				Filename:             "test.proto",
+				Template:             tmpl,
+				Data:                 test,
+			},
+		)
+		reg, err := lint.MakeRegistryFromAllFiles([]*descriptorpb.FileDescriptorProto{fdp})
```

Comment:
We shouldn't expose this method just for the sake of this test. This functionality doesn't really even belong in the linter – I wrote it as a temporary helper until the proto team adds this feature to protodesc. Try using [protoregistry.NewFiles](https://github.com/golang/protobuf/blob/api-v2/reflect/protoregistry/registry.go#L76) directly.

Maybe you can make a new `testutil` method that takes a directory path and outputs a `[]*descriptorpb.FileDescriptorProto`. Or maybe there's some way to get the MethodDescriptor from the descriptorpb version of it.

Reply:
`protoregistry.NewFiles` takes a `protoreflect.FileDescriptor`, which is not the thing I have.

Reply:
Yeah, the coordination between the two packages can be a bit annoying, but I still feel that this function should not be exposed from our package. We intend on removing it. I would almost prefer you copy paste it as a private function into this package if you can't find a cleaner way to test your helper.